When deleting an `azurerm_data_protection_backup_vault` with its resource group, Azure reports the resource group isn't empty

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.13.5

### AzureRM Provider Version

4.54.0

### Affected Resource(s)/Data Source(s)

azurerm_data_protection_backup_vault

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    azurerm = {
      source = "registry.terraform.io/hashicorp/azurerm"
    }
  }
}

provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "test" {
  name     = "acctest-dataprotection-251122161058142591"
  location = "Canada Central"
}

resource "azurerm_data_protection_backup_vault" "test" {
  name                = "acctest-dbv-251122161058142591"
  resource_group_name = azurerm_resource_group.test.name
  location            = azurerm_resource_group.test.location
  datastore_type      = "VaultStore"
  redundancy          = "LocallyRedundant"
}

resource "azurerm_data_protection_backup_policy_blob_storage" "test" {
  name                                   = "acctest-dbp-251122161058142591"
  vault_id                               = azurerm_data_protection_backup_vault.test.id
  operational_default_retention_duration = "P30D"
  vault_default_retention_duration       = "P7D"
  backup_repeating_time_intervals        = ["R/2024-05-08T11:30:00+00:00/P1W"]

  retention_rule {
    name     = "Monthly"
    priority = 15
    life_cycle {
      duration        = "P6M"
      data_store_type = "VaultStore"
    }
    criteria {
      days_of_month = [1, 2, 0]
    }
  }

  retention_rule {
    name     = "Daily"
    priority = 25
    life_cycle {
      duration        = "P7D"
      data_store_type = "VaultStore"
    }
    criteria {
      days_of_week           = ["Thursday"]
      months_of_year         = ["November"]
      weeks_of_month         = ["First"]
      scheduled_backup_times = ["2024-05-08T02:30:00Z"]
    }
  }
}
```

### Debug Output/Panic Output

```shell
Error: deleting Resource Group "acctest-dataprotection-251122161058142591": the Resource Group still contains Resources.

Terraform is configured to check for Resources within the Resource Group when deleting the Resource Group - and
raise an error if nested Resources still exist to avoid unintentionally deleting these Resources.

Terraform has detected that the following Resources still exist within the Resource Group:

* `/subscriptions/<sub_id>/resourceGroups/acctest-dataprotection-251122161058142591/providers/Microsoft.DataProtection/BackupVaults/acctest-dbv-251122161058142591`

This feature is intended to avoid the unintentional destruction of nested Resources provisioned through some
other means (for example, an ARM Template Deployment) - as such you must either remove these Resources, or
disable this behaviour using the feature flag `prevent_deletion_if_contains_resources` within the `features`
block when configuring the Provider, for example:

provider "azurerm" {
  features {
    resource_group {
      prevent_deletion_if_contains_resources = false
    }
  }
}

When that feature flag is set, Terraform will skip checking for any Resources within the Resource Group and
delete this using the Azure API directly (which will clear up any nested resources).

More information on the `features` block can be found in the documentation:
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/features-block
```

### Expected Behaviour

When deleting/destroying the configuration, deletion should succeed.

### Actual Behaviour

See error.

I have been working on some stuff in the provider (see: #31107, #31170, #31109) and something has changed in the past couple of weeks. Acceptance tests that were passing are now failing any time an `azurerm_data_protection_backup_vault` is used with the attached error.

Looking at the logs, we see:

```bash
2025-11-26T15:35:17.649Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: [DEBUG] AzureRM Request: 
2025-11-26T15:35:17.649Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: DELETE /subscriptions/<sub_guid>/resourceGroups/acctest-dataprotection-251122161058142591/providers/Microsoft.DataProtection/backupVaults/acctest-dbv-251122161058142591/backupPolicies/acctest-dbp-251122161058142591?api-version=2024-04-01 HTTP/1.1
2025-11-26T15:35:17.649Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Host: management.azure.com
2025-11-26T15:35:17.649Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: User-Agent: HashiCorp/go-azure-sdk (Go-http-Client/1.1 backuppolicies/2024-04-01) HashiCorp Terraform/1.13.5 (+https://www.terraform.io) terraform-provider-azurerm/dev pid-222c6c49-1b0a-5959-a213-6608f9eb8820
2025-11-26T15:35:17.649Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Content-Type: application/json; charset=utf-8
2025-11-26T15:35:17.649Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Ms-Correlation-Request-Id: 5ac9623b-e6b7-c56d-5db5-efd7fa090c7c
2025-11-26T15:35:17.649Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Accept-Encoding: gzip
2025-11-26T15:35:17.649Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5
2025-11-26T15:35:17.649Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5
2025-11-26T15:35:17.649Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: [DEBUG] DELETE https://management.azure.com/subscriptions/<sub_guid>/resourceGroups/acctest-dataprotection-251122161058142591/providers/Microsoft.DataProtection/backupVaults/acctest-dbv-251122161058142591/backupPolicies/acctest-dbp-251122161058142591?api-version=2024-04-01
2025-11-26T15:35:17.363Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: [DEBUG] AzureRM Response for https://management.azure.com/subscriptions/<sub_guid>/resourceGroups/acctest-dataprotection-251122161058142591/providers/Microsoft.DataProtection/backupVaults/acctest-dbv-251122161058142591/backupPolicies/acctest-dbp-251122161058142591?api-version=2024-04-01: 
2025-11-26T15:35:17.363Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: HTTP/2.0 200 OK
2025-11-26T15:35:17.363Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Cache-Control: no-cache
2025-11-26T15:35:17.363Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Date: Wed, 26 Nov 2025 15:35:17 GMT
2025-11-26T15:35:17.363Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Expires: -1
2025-11-26T15:35:17.363Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Pragma: no-cache
2025-11-26T15:35:17.363Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Strict-Transport-Security: max-age=31536000; includeSubDomains
2025-11-26T15:35:17.363Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Cache: CONFIG_NOCACHE
2025-11-26T15:35:17.363Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Content-Type-Options: nosniff
2025-11-26T15:35:17.363Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Ms-Correlation-Request-Id: 5ac9623b-e6b7-c56d-5db5-efd7fa090c7c
2025-11-26T15:35:17.363Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Ms-Operation-Identifier: appId=bcdae85a-6fc4-45f7-abf0-1cbdd6bd015a,tenantId=<tenant_guid>,objectId=b788ac66-1dbc-4744-9f03-6f6585727daf/canadacentral/9e5b6d9c-9556-43a0-805f-b2c8a6c839e1
2025-11-26T15:35:17.363Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Ms-Ratelimit-Remaining-Subscription-Resource-Requests: 799
2025-11-26T15:35:17.363Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Ms-Request-Id: 33a32246-212e-44ed-a0e7-09db768ff931
2025-11-26T15:35:17.363Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Ms-Routing-Request-Id: CANADACENTRAL:20251126T153517Z:33a32246-212e-44ed-a0e7-09db768ff931
2025-11-26T15:35:17.363Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Msedge-Ref: Ref A: 7A6D0B569AAA4F20A8BF44E47BF5AF0C Ref B: CO6AA3150217039 Ref C: 2025-11-26T15:35:16Z
2025-11-26T15:35:17.363Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Content-Length: 0
2025-11-26T15:35:17.363Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5
2025-11-26T15:35:17.363Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5
2025-11-26T15:35:17.366Z [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2025-11-26T15:35:17.367Z [INFO]  Starting apply for azurerm_data_protection_backup_vault.test
2025-11-26T15:35:17.367Z [DEBUG] azurerm_data_protection_backup_vault.test: applying the planned Delete change
2025-11-26T15:35:17.368Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: [DEBUG] AzureRM Request: 
2025-11-26T15:35:17.368Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: DELETE /subscriptions/<sub_guid>/resourceGroups/acctest-dataprotection-251122161058142591/providers/Microsoft.DataProtection/backupVaults/acctest-dbv-251122161058142591?api-version=2024-04-01 HTTP/1.1
2025-11-26T15:35:17.368Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Host: management.azure.com
2025-11-26T15:35:17.368Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: User-Agent: HashiCorp/go-azure-sdk (Go-http-Client/1.1 backupvaults/2024-04-01) HashiCorp Terraform/1.13.5 (+https://www.terraform.io) terraform-provider-azurerm/dev pid-222c6c49-1b0a-5959-a213-6608f9eb8820
2025-11-26T15:35:17.368Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Content-Type: application/json; charset=utf-8
2025-11-26T15:35:17.368Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Ms-Correlation-Request-Id: 5ac9623b-e6b7-c56d-5db5-efd7fa090c7c
2025-11-26T15:35:17.368Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Accept-Encoding: gzip
2025-11-26T15:35:17.368Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5
2025-11-26T15:35:17.368Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5
***2025-11-26T15:35:17.368Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: [DEBUG] DELETE https://management.azure.com/subscriptions/<sub_guid>/resourceGroups/acctest-dataprotection-251122161058142591/providers/Microsoft.DataProtection/backupVaults/acctest-dbv-251122161058142591?api-version=2024-04-01
2025-11-26T15:35:18.023Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: [DEBUG] AzureRM Response for https://management.azure.com/subscriptions/<sub_guid>/resourceGroups/acctest-dataprotection-251122161058142591/providers/Microsoft.DataProtection/backupVaults/acctest-dbv-251122161058142591?api-version=2024-04-01: ***
2025-11-26T15:35:18.023Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: HTTP/2.0 202 Accepted
2025-11-26T15:35:18.023Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Azure-Asyncoperation: https://management.azure.com/subscriptions/<sub_guid>/resourceGroups/acctest-dataprotection-251122161058142591/providers/Microsoft.DataProtection/locations/canadacentral/operationStatus/N2VlNTJiOWItOGNlMi00NGJiLThlODgtOGFlMmM2MmVlYzQyOzBlYjM5MGI5LTJiZjgtNDNhOS04MDAyLTc2YTIwOWRhMzQyNg==?api-version=2024-04-01&t=638997681179231706&c=MIIHpTCCBo2gAwIBAgITfwY2O9Fi5lSmLyOBKgAEBjY70TANBgkqhkiG9w0BAQsFADBEMRMwEQYKCZImiZPyLGQBGRYDR0JMMRMwEQYKCZImiZPyLGQBGRYDQU1FMRgwFgYDVQQDEw9BTUUgSW5mcmEgQ0EgMDIwHhcNMjUxMDIwMDQxMjEwWhcNMjYwNDE4MDQxMjEwWjBAMT4wPAYDVQQDEzVhc3luY29wZXJhdGlvbnNpZ25pbmdjZXJ0aWZpY2F0ZS5tYW5hZ2VtZW50LmF6dXJlLmNvbTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBAMHQm7ZXwmvz3bp_tBu0ZYNZMpMcRUJSHrld-E9b4RqS0ymmswmhM267lbAI2oCZNI82E4DZNNZApPYz9GqZQ2OYxE0G0E_dyvstulti8jSM29pm5HiR_hxNyzt_q0uSnmqQnWuxQMco30qz4XjZaQkJNBATBraJWnioUZtMKhrbXrMOMral9bW_6ba5EqNlc3iBHJ8x7dblRrVcao1gyCk1mEXz9vR8Ut0wZ-KLWuSUcbcS7CPzUuSYQ1eJdym4YYAqEQQM1z40lpqT6OlbqRLu3DC5dN2NbGHUPkHODMhUbKY3RnwHKdrKQbuDk16nRBjl2ZML6kXZGz_BMd-elV0CAwEAAaOCBJIwggSOMCcGCSsGAQQBgjcVCgQaMBgwCgYIKwYBBQUHAwEwCgYIKwYBBQUHAwIwPQYJKwYBBAGCNxUHBDAwLgYmKwYBBAGCNxUIhpDjDYTVtHiE8Ys-hZvdFs6dEoFghfmRS4WsmTQCAWQCAQcwggHaBggrBgEFBQcBAQSCAcwwggHIMGYGCCsGAQUFBzAChlpodHRwOi8vY3JsLm1pY3Jvc29mdC5jb20vcGtpaW5mcmEvQ2VydHMvQkwyUEtJSU5UQ0EwMS5BTUUuR0JMX0FNRSUyMEluZnJhJTIwQ0ElMjAwMig0KS5jcnQwVgYIKwYBBQUHMAKGSmh0dHA6Ly9jcmwxLmFtZS5nYmwvYWlhL0JMMlBLSUlOVENBMDEuQU1FLkdCTF9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3J0MFYGCCsGAQUFBzAChkpodHRwOi8vY3JsMi5hbWUuZ2JsL2FpYS9CTDJQS0lJTlRDQTAxLkFNRS5HQkxfQU1FJTIwSW5mcmElMjBDQSUyMDAyKDQpLmNydDBWBggrBgEFBQcwAoZKaHR0cDovL2NybDMuYW1lLmdibC9haWEvQkwyUEtJSU5UQ0EwMS5BTUUuR0JMX0FNRSUyMEluZnJhJTIwQ0ElMjAwMig0KS5jcnQwVgYIKwYBBQUHMAKGSmh0dHA6Ly9jcmw0LmFtZS5nYmwvYWlhL0JMMlBLSUlOVENBMDEuQU1FLkdCTF9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3J0MB0GA1UdDgQWBBQ7Uo6SICZvdsfQb4hxJfOTmnlMAzAOBgNVHQ8BAf8EBAMCBaAwggE1BgNVHR8EggEsMIIBKDCCASSgggEgoIIBHIZCaHR0cDovL2NybC5taWNyb3NvZnQuY29tL3BraWluZnJhL0NSTC9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3JshjRodHRwOi8vY3JsMS5hbWUuZ2JsL2NybC9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3JshjRodHRwOi8vY3JsMi5hbWUuZ2JsL2NybC9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3JshjRodHRwOi8vY3JsMy5hbWUuZ2JsL2NybC9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3JshjRodHRwOi8vY3JsNC5hbWUuZ2JsL2NybC9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3JsMIGdBgNVHSAEgZUwgZIwDAYKKwYBBAGCN3sBATBmBgorBgEEAYI3ewICMFgwVgYIKwYBBQUHAgIwSh5IADMAMwBlADAAMQA5ADIAMQAtADQAZAA2ADQALQA0AGYAOABjAC0AYQAwADUANQAtADUAYgBkAGEAZgBmAGQANQBlADMAMwBkMAwGCisGAQQBgjd7AwIwDAYKKwYBBAGCN3sEAjAfBgNVHSMEGDAWgBSuecJrXSWIEwb2BwnDl3x7l48dVTAdBgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIwDQYJKoZIhvcNAQELBQADggEBADp-OzkMAESDX27pXk3NL5_uU6RuCBkTjHnVxyWwp89arbWDWeL0FNU75N4hZMsK4FU9oUeatHrGk09j6u3o6qgzbyH7vWX9v5ijaUpNPeacRaMTpWvVZZLg6AERFF7oiplb-vCmLHZTMYbJH6dZI35unEEDgzD9n8QhuMBQ8dXhS145FHrAVc9nnNB25FQt29UfdpwrVFPQuEkgwKKCG8Qy5VDtGnceMRw2bnDwtYNngCgVmd2cPtwgizpflj_F8NURyxhnrQvuOwCkQavcJTq5RSXgEp3U471r3TZoEkKrAmcNF3Fyz5M0gOv6EQmdHg-lLMeFUmOVsXvWTlMjQQw&s=YC83G4a6zcjU2dgw2kDabvtmu2xewfIug3530qMZNxhghH06OrATrpHPWdPMIUWygd8oARliuzvsCLRv9EUetHefcrnRoSayylkNbAwsUkkLx7YgW-xG01LXe97xdXV4UGuDKqVwPaeDOKDKzfmfCF5cIyq0_IjRCzgmX67-85J-4URURwgvfZG7NO0IezW2VMCP1hZR6Bm_BlPrxFoHwHl00FaxzbtxY-X0hqgLE8iykLlTUbtzwE1den-qD9vk7eHqoA9ysnerRGzfb5r7YJPIthx6UL2MdmOV19z2OorfJQrPLakFgzhHecGpysRFvboL7w_IRWgesp75HCeqJQ&h=rCTIowGUVmzGdZvW-kz_Fkn6yRHFrrk7wKdtttqY-m4
2025-11-26T15:35:18.023Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Cache-Control: no-cache
2025-11-26T15:35:18.023Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Date: Wed, 26 Nov 2025 15:35:17 GMT
2025-11-26T15:35:18.023Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Expires: -1
2025-11-26T15:35:18.023Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Location: https://management.azure.com/subscriptions/<sub_guid>/providers/Microsoft.DataProtection/locations/canadacentral/operationResults/N2VlNTJiOWItOGNlMi00NGJiLThlODgtOGFlMmM2MmVlYzQyOzBlYjM5MGI5LTJiZjgtNDNhOS04MDAyLTc2YTIwOWRhMzQyNg==?api-version=2024-04-01&t=638997681179387901&c=MIIHpTCCBo2gAwIBAgITfwY2O9Fi5lSmLyOBKgAEBjY70TANBgkqhkiG9w0BAQsFADBEMRMwEQYKCZImiZPyLGQBGRYDR0JMMRMwEQYKCZImiZPyLGQBGRYDQU1FMRgwFgYDVQQDEw9BTUUgSW5mcmEgQ0EgMDIwHhcNMjUxMDIwMDQxMjEwWhcNMjYwNDE4MDQxMjEwWjBAMT4wPAYDVQQDEzVhc3luY29wZXJhdGlvbnNpZ25pbmdjZXJ0aWZpY2F0ZS5tYW5hZ2VtZW50LmF6dXJlLmNvbTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBAMHQm7ZXwmvz3bp_tBu0ZYNZMpMcRUJSHrld-E9b4RqS0ymmswmhM267lbAI2oCZNI82E4DZNNZApPYz9GqZQ2OYxE0G0E_dyvstulti8jSM29pm5HiR_hxNyzt_q0uSnmqQnWuxQMco30qz4XjZaQkJNBATBraJWnioUZtMKhrbXrMOMral9bW_6ba5EqNlc3iBHJ8x7dblRrVcao1gyCk1mEXz9vR8Ut0wZ-KLWuSUcbcS7CPzUuSYQ1eJdym4YYAqEQQM1z40lpqT6OlbqRLu3DC5dN2NbGHUPkHODMhUbKY3RnwHKdrKQbuDk16nRBjl2ZML6kXZGz_BMd-elV0CAwEAAaOCBJIwggSOMCcGCSsGAQQBgjcVCgQaMBgwCgYIKwYBBQUHAwEwCgYIKwYBBQUHAwIwPQYJKwYBBAGCNxUHBDAwLgYmKwYBBAGCNxUIhpDjDYTVtHiE8Ys-hZvdFs6dEoFghfmRS4WsmTQCAWQCAQcwggHaBggrBgEFBQcBAQSCAcwwggHIMGYGCCsGAQUFBzAChlpodHRwOi8vY3JsLm1pY3Jvc29mdC5jb20vcGtpaW5mcmEvQ2VydHMvQkwyUEtJSU5UQ0EwMS5BTUUuR0JMX0FNRSUyMEluZnJhJTIwQ0ElMjAwMig0KS5jcnQwVgYIKwYBBQUHMAKGSmh0dHA6Ly9jcmwxLmFtZS5nYmwvYWlhL0JMMlBLSUlOVENBMDEuQU1FLkdCTF9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3J0MFYGCCsGAQUFBzAChkpodHRwOi8vY3JsMi5hbWUuZ2JsL2FpYS9CTDJQS0lJTlRDQTAxLkFNRS5HQkxfQU1FJTIwSW5mcmElMjBDQSUyMDAyKDQpLmNydDBWBggrBgEFBQcwAoZKaHR0cDovL2NybDMuYW1lLmdibC9haWEvQkwyUEtJSU5UQ0EwMS5BTUUuR0JMX0FNRSUyMEluZnJhJTIwQ0ElMjAwMig0KS5jcnQwVgYIKwYBBQUHMAKGSmh0dHA6Ly9jcmw0LmFtZS5nYmwvYWlhL0JMMlBLSUlOVENBMDEuQU1FLkdCTF9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3J0MB0GA1UdDgQWBBQ7Uo6SICZvdsfQb4hxJfOTmnlMAzAOBgNVHQ8BAf8EBAMCBaAwggE1BgNVHR8EggEsMIIBKDCCASSgggEgoIIBHIZCaHR0cDovL2NybC5taWNyb3NvZnQuY29tL3BraWluZnJhL0NSTC9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3JshjRodHRwOi8vY3JsMS5hbWUuZ2JsL2NybC9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3JshjRodHRwOi8vY3JsMi5hbWUuZ2JsL2NybC9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3JshjRodHRwOi8vY3JsMy5hbWUuZ2JsL2NybC9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3JshjRodHRwOi8vY3JsNC5hbWUuZ2JsL2NybC9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3JsMIGdBgNVHSAEgZUwgZIwDAYKKwYBBAGCN3sBATBmBgorBgEEAYI3ewICMFgwVgYIKwYBBQUHAgIwSh5IADMAMwBlADAAMQA5ADIAMQAtADQAZAA2ADQALQA0AGYAOABjAC0AYQAwADUANQAtADUAYgBkAGEAZgBmAGQANQBlADMAMwBkMAwGCisGAQQBgjd7AwIwDAYKKwYBBAGCN3sEAjAfBgNVHSMEGDAWgBSuecJrXSWIEwb2BwnDl3x7l48dVTAdBgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIwDQYJKoZIhvcNAQELBQADggEBADp-OzkMAESDX27pXk3NL5_uU6RuCBkTjHnVxyWwp89arbWDWeL0FNU75N4hZMsK4FU9oUeatHrGk09j6u3o6qgzbyH7vWX9v5ijaUpNPeacRaMTpWvVZZLg6AERFF7oiplb-vCmLHZTMYbJH6dZI35unEEDgzD9n8QhuMBQ8dXhS145FHrAVc9nnNB25FQt29UfdpwrVFPQuEkgwKKCG8Qy5VDtGnceMRw2bnDwtYNngCgVmd2cPtwgizpflj_F8NURyxhnrQvuOwCkQavcJTq5RSXgEp3U471r3TZoEkKrAmcNF3Fyz5M0gOv6EQmdHg-lLMeFUmOVsXvWTlMjQQw&s=lHhcK-tIwYl8ehED7l0MmafuxvugEuQZv2oSQVXCk3chpvCoaJ-Jbyisyotm6ZlxU1hp0GHmqRfp4ZThH6xzGGI0IsFBnHNScbDn7nRzDS__x5cqZ_s9srx5hxBu-5cWOJaMv6RNOt51y9ZFeHkl5ViU32GWV00WNhoRY9ssIEs1mZaRLWsO3m7xOlkJjiCyNlxDRn0IICLdQ9mKamuI4JsT38oe8gAtOjURxKuxR7GlfPQwTLUDdrp2HVJalFz3dc0EycEF3J7xy2C0Us7_BMT0kQ1UuTZkC6jsO16hxEiXlF5s1q64wogf5nsro6bl5MV-lNW21_3xGwiRLvTr2g&h=qozxhN_oD09JKmskWmFZuZ8iDHnFxfVcEXlHhcfRa-A
2025-11-26T15:35:18.023Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Pragma: no-cache
2025-11-26T15:35:18.023Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Retry-After: 30
2025-11-26T15:35:18.023Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Strict-Transport-Security: max-age=31536000; includeSubDomains
2025-11-26T15:35:18.023Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Cache: CONFIG_NOCACHE
2025-11-26T15:35:18.023Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Content-Type-Options: nosniff
2025-11-26T15:35:18.024Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Ms-Correlation-Request-Id: 5ac9623b-e6b7-c56d-5db5-efd7fa090c7c
2025-11-26T15:35:18.024Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Ms-Operation-Identifier: appId=bcdae85a-6fc4-45f7-abf0-1cbdd6bd015a,tenantId=<tenant_guid>,objectId=b788ac66-1dbc-4744-9f03-6f6585727daf/canadacentral/5e1651b0-7e4e-4e92-809a-dbf01a288a58
2025-11-26T15:35:18.024Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Ms-Ratelimit-Remaining-Subscription-Resource-Requests: 799
2025-11-26T15:35:18.024Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Ms-Request-Id: 92865673-91f5-47a7-b708-d2c7f9ce6b00
2025-11-26T15:35:18.024Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Ms-Routing-Request-Id: CANADACENTRAL:20251126T153517Z:92865673-91f5-47a7-b708-d2c7f9ce6b00
2025-11-26T15:35:18.024Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Msedge-Ref: Ref A: 76CCC32D963A482A856B4B14E2102856 Ref B: CO6AA3150217047 Ref C: 2025-11-26T15:35:17Z
2025-11-26T15:35:18.024Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Content-Length: 0
2025-11-26T15:35:18.024Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5
2025-11-26T15:35:18.024Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5
2025-11-26T15:35:18.026Z [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2025-11-26T15:35:18.027Z [INFO]  Starting apply for azurerm_resource_group.test
2025-11-26T15:35:18.027Z [DEBUG] azurerm_resource_group.test: applying the planned Delete change
2025-11-26T15:35:28.037Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: [DEBUG] AzureRM Request: 
2025-11-26T15:35:28.037Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: GET /subscriptions/<sub_guid>/resourceGroups/acctest-dataprotection-251122161058142591/resources?%24expand=provisioningState&%24top=10&api-version=2023-07-01 HTTP/1.1
2025-11-26T15:35:28.037Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Host: management.azure.com
2025-11-26T15:35:28.037Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: User-Agent: HashiCorp/go-azure-sdk (Go-http-Client/1.1 resourcegroups/2023-07-01) HashiCorp Terraform/1.13.5 (+https://www.terraform.io) terraform-provider-azurerm/dev pid-222c6c49-1b0a-5959-a213-6608f9eb8820
2025-11-26T15:35:28.037Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Accept: application/json; charset=utf-8; IEEE754Compatible=false
2025-11-26T15:35:28.037Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Content-Type: application/json; charset=utf-8
2025-11-26T15:35:28.037Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Odata-Maxversion: 4.0
2025-11-26T15:35:28.037Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Odata-Version: 4.0
2025-11-26T15:35:28.037Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Accept-Encoding: gzip
2025-11-26T15:35:28.037Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5
2025-11-26T15:35:28.037Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5
2025-11-26T15:35:28.037Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: [DEBUG] GET https://management.azure.com/subscriptions/<sub_guid>/resourceGroups/acctest-dataprotection-251122161058142591/resources?%24expand=provisioningState&%24top=10&api-version=2023-07-01
2025-11-26T15:35:28.419Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: [DEBUG] AzureRM Response for https://management.azure.com/subscriptions/<sub_guid>/resourceGroups/acctest-dataprotection-251122161058142591/resources?%24expand=provisioningState&%24top=10&api-version=2023-07-01: 
2025-11-26T15:35:28.419Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: HTTP/2.0 200 OK
2025-11-26T15:35:28.419Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Content-Length: 391
2025-11-26T15:35:28.419Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Cache-Control: no-cache
2025-11-26T15:35:28.419Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Content-Type: application/json; charset=utf-8
2025-11-26T15:35:28.419Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Date: Wed, 26 Nov 2025 15:35:27 GMT
2025-11-26T15:35:28.419Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Expires: -1
2025-11-26T15:35:28.419Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Pragma: no-cache
2025-11-26T15:35:28.419Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: Strict-Transport-Security: max-age=31536000; includeSubDomains
2025-11-26T15:35:28.419Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Cache: CONFIG_NOCACHE
2025-11-26T15:35:28.419Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Content-Type-Options: nosniff
2025-11-26T15:35:28.419Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Ms-Correlation-Request-Id: 618e9f92-634f-4f70-ae1e-a8e6a845e1b7
2025-11-26T15:35:28.420Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Ms-Ratelimit-Remaining-Subscription-Global-Reads: 3749
2025-11-26T15:35:28.420Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Ms-Ratelimit-Remaining-Subscription-Reads: 249
2025-11-26T15:35:28.420Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Ms-Request-Id: 618e9f92-634f-4f70-ae1e-a8e6a845e1b7
2025-11-26T15:35:28.420Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Ms-Routing-Request-Id: CANADACENTRAL:20251126T153527Z:618e9f92-634f-4f70-ae1e-a8e6a845e1b7
2025-11-26T15:35:28.420Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: X-Msedge-Ref: Ref A: A5119D553EE64D6AB37EF864611EBBC0 Ref B: MWH011020806042 Ref C: 2025-11-26T15:35:27Z
2025-11-26T15:35:28.420Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5
2025-11-26T15:35:28.420Z [DEBUG] provider.terraform-provider-azurerm_v4.54.0_x5: {"value":[{"id":"/subscriptions/<sub_guid>/resourceGroups/acctest-dataprotection-251122161058142591/providers/Microsoft.DataProtection/BackupVaults/acctest-dbv-251122161058142591","name":"acctest-dbv-251122161058142591","type":"Microsoft.DataProtection/BackupVaults","location":"canadacentral","identity":{"type":"None"},"provisioningState":"Succeeded","tags":{}}]}
2025-11-26T15:35:28.420Z [ERROR] provider.terraform-provider-azurerm_v4.54.0_x5: Response contains error diagnostic: tf_proto_version=5.10 tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_req_id=dbf23961-08fa-4c92-2146-0ccf4a10f2dd tf_resource_type=azurerm_resource_group @caller=github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov5/internal/diag/diagnostics.go:58 @module=sdk.proto diagnostic_detail="" diagnostic_severity=ERROR
  diagnostic_summary=
  | deleting Resource Group "acctest-dataprotection-251122161058142591": the Resource Group still contains Resources.
  | 
  | Terraform is configured to check for Resources within the Resource Group when deleting the Resource Group - and
  | raise an error if nested Resources still exist to avoid unintentionally deleting these Resources.
  | 
  | Terraform has detected that the following Resources still exist within the Resource Group:
  | 
  | * `/subscriptions/<sub_guid>/resourceGroups/acctest-dataprotection-251122161058142591/providers/Microsoft.DataProtection/BackupVaults/acctest-dbv-251122161058142591`
  | 
  | This feature is intended to avoid the unintentional destruction of nested Resources provisioned through some
  | other means (for example, an ARM Template Deployment) - as such you must either remove these Resources, or
  | disable this behaviour using the feature flag `prevent_deletion_if_contains_resources` within the `features`
  | block when configuring the Provider, for example:
  | 
  | provider "azurerm" {
  |   features {
  |     resource_group {
  |       prevent_deletion_if_contains_resources = false
  |     }
  |   }
  | }
  | 
  | When that feature flag is set, Terraform will skip checking for any Resources within the Resource Group and
  | delete this using the Azure API directly (which will clear up any nested resources).
  | 
  | More information on the `features` block can be found in the documentation:
  | https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/features-block
   tf_rpc=ApplyResourceChange timestamp=2025-11-26T15:35:28.420Z

```

Note the `202 Accepted` response from Azure to delete the vault.

There's actually a bug here which I can submit a PR for; [this line](https://github.com/hashicorp/terraform-provider-azurerm/blob/005c4496c5df57f6ff66b1f29cb9956b28ad4256/internal/services/dataprotection/data_protection_backup_vault_resource.go#L296) and the few after it should actually be:

```golang
if err := client.DeleteThenPoll(ctx, *id); err != nil {
	return fmt.Errorf("deleting DataProtection BackupVault (%q): %+v", id, err)
}
```

(e.g., the method should be `DeleteThenPoll`, not `Delete`); HOWEVER, even when making that change and trying again, this fails.

Logs from a run using that change:

```bash
DELETE /subscriptions/<sub_guid>/resourceGroups/acctest-dataprotection-251125213904888367/providers/Microsoft.DataProtection/backupVaults/acctest-dbv-251125213904888367?api-version=2024-04-01 HTTP/1.1
Host: management.azure.com
User-Agent: HashiCorp/go-azure-sdk (Go-http-Client/1.1 backupvaults/2024-04-01) HashiCorp Terraform/1.13.5 (+https://www.terraform.io) terraform-provider-azurerm/acc pid-222c6c49-1b0a-5959-a213-6608f9eb8820
Content-Type: application/json; charset=utf-8
X-Ms-Correlation-Request-Id: 909b11be-8d7c-7deb-e98e-e2ff08be4143
Accept-Encoding: gzip


2025/11/25 21:40:14 [DEBUG] DELETE https://management.azure.com/subscriptions/<sub_guid>/resourceGroups/acctest-dataprotection-251125213904888367/providers/Microsoft.DataProtection/backupVaults/acctest-dbv-251125213904888367?api-version=2024-04-01
2025/11/25 21:40:14 [DEBUG] AzureRM Response for https://management.azure.com/subscriptions/<sub_guid>/resourceGroups/acctest-dataprotection-251125213904888367/providers/Microsoft.DataProtection/backupVaults/acctest-dbv-251125213904888367?api-version=2024-04-01: 
HTTP/2.0 202 Accepted
Azure-Asyncoperation: https://management.azure.com/subscriptions/<sub_guid>/resourceGroups/acctest-dataprotection-251125213904888367/providers/Microsoft.DataProtection/locations/canadacentral/operationStatus/NDA1ODVhNWEtYjk3Ny00ZmI5LTlhYWYtNjIwNTI3ODJiYjFkO2VjMDcwOWE2LWRmOTktNDUxZi1iMDA4LWQ4MDQxYjdlNWI4Nw==?api-version=2024-04-01&t=638997036148195543&c=MIIHpTCCBo2gAwIBAgITfwY2O9Fi5lSmLyOBKgAEBjY70TANBgkqhkiG9w0BAQsFADBEMRMwEQYKCZImiZPyLGQBGRYDR0JMMRMwEQYKCZImiZPyLGQBGRYDQU1FMRgwFgYDVQQDEw9BTUUgSW5mcmEgQ0EgMDIwHhcNMjUxMDIwMDQxMjEwWhcNMjYwNDE4MDQxMjEwWjBAMT4wPAYDVQQDEzVhc3luY29wZXJhdGlvbnNpZ25pbmdjZXJ0aWZpY2F0ZS5tYW5hZ2VtZW50LmF6dXJlLmNvbTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBAMHQm7ZXwmvz3bp_tBu0ZYNZMpMcRUJSHrld-E9b4RqS0ymmswmhM267lbAI2oCZNI82E4DZNNZApPYz9GqZQ2OYxE0G0E_dyvstulti8jSM29pm5HiR_hxNyzt_q0uSnmqQnWuxQMco30qz4XjZaQkJNBATBraJWnioUZtMKhrbXrMOMral9bW_6ba5EqNlc3iBHJ8x7dblRrVcao1gyCk1mEXz9vR8Ut0wZ-KLWuSUcbcS7CPzUuSYQ1eJdym4YYAqEQQM1z40lpqT6OlbqRLu3DC5dN2NbGHUPkHODMhUbKY3RnwHKdrKQbuDk16nRBjl2ZML6kXZGz_BMd-elV0CAwEAAaOCBJIwggSOMCcGCSsGAQQBgjcVCgQaMBgwCgYIKwYBBQUHAwEwCgYIKwYBBQUHAwIwPQYJKwYBBAGCNxUHBDAwLgYmKwYBBAGCNxUIhpDjDYTVtHiE8Ys-hZvdFs6dEoFghfmRS4WsmTQCAWQCAQcwggHaBggrBgEFBQcBAQSCAcwwggHIMGYGCCsGAQUFBzAChlpodHRwOi8vY3JsLm1pY3Jvc29mdC5jb20vcGtpaW5mcmEvQ2VydHMvQkwyUEtJSU5UQ0EwMS5BTUUuR0JMX0FNRSUyMEluZnJhJTIwQ0ElMjAwMig0KS5jcnQwVgYIKwYBBQUHMAKGSmh0dHA6Ly9jcmwxLmFtZS5nYmwvYWlhL0JMMlBLSUlOVENBMDEuQU1FLkdCTF9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3J0MFYGCCsGAQUFBzAChkpodHRwOi8vY3JsMi5hbWUuZ2JsL2FpYS9CTDJQS0lJTlRDQTAxLkFNRS5HQkxfQU1FJTIwSW5mcmElMjBDQSUyMDAyKDQpLmNydDBWBggrBgEFBQcwAoZKaHR0cDovL2NybDMuYW1lLmdibC9haWEvQkwyUEtJSU5UQ0EwMS5BTUUuR0JMX0FNRSUyMEluZnJhJTIwQ0ElMjAwMig0KS5jcnQwVgYIKwYBBQUHMAKGSmh0dHA6Ly9jcmw0LmFtZS5nYmwvYWlhL0JMMlBLSUlOVENBMDEuQU1FLkdCTF9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3J0MB0GA1UdDgQWBBQ7Uo6SICZvdsfQb4hxJfOTmnlMAzAOBgNVHQ8BAf8EBAMCBaAwggE1BgNVHR8EggEsMIIBKDCCASSgggEgoIIBHIZCaHR0cDovL2NybC5taWNyb3NvZnQuY29tL3BraWluZnJhL0NSTC9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3JshjRodHRwOi8vY3JsMS5hbWUuZ2JsL2NybC9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3JshjRodHRwOi8vY3JsMi5hbWUuZ2JsL2NybC9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3JshjRodHRwOi8vY3JsMy5hbWUuZ2JsL2NybC9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3JshjRodHRwOi8vY3JsNC5hbWUuZ2JsL2NybC9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3JsMIGdBgNVHSAEgZUwgZIwDAYKKwYBBAGCN3sBATBmBgorBgEEAYI3ewICMFgwVgYIKwYBBQUHAgIwSh5IADMAMwBlADAAMQA5ADIAMQAtADQAZAA2ADQALQA0AGYAOABjAC0AYQAwADUANQAtADUAYgBkAGEAZgBmAGQANQBlADMAMwBkMAwGCisGAQQBgjd7AwIwDAYKKwYBBAGCN3sEAjAfBgNVHSMEGDAWgBSuecJrXSWIEwb2BwnDl3x7l48dVTAdBgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIwDQYJKoZIhvcNAQELBQADggEBADp-OzkMAESDX27pXk3NL5_uU6RuCBkTjHnVxyWwp89arbWDWeL0FNU75N4hZMsK4FU9oUeatHrGk09j6u3o6qgzbyH7vWX9v5ijaUpNPeacRaMTpWvVZZLg6AERFF7oiplb-vCmLHZTMYbJH6dZI35unEEDgzD9n8QhuMBQ8dXhS145FHrAVc9nnNB25FQt29UfdpwrVFPQuEkgwKKCG8Qy5VDtGnceMRw2bnDwtYNngCgVmd2cPtwgizpflj_F8NURyxhnrQvuOwCkQavcJTq5RSXgEp3U471r3TZoEkKrAmcNF3Fyz5M0gOv6EQmdHg-lLMeFUmOVsXvWTlMjQQw&s=Gy3HbeitOfeFITPVEWhFCh8lOv_Q0p7AUG1YEjfwWnV3HEuCw6oTTtS0aYI5CQ89K0l5VJgxxJ-Jd80jJxbUql7YINBCYHCu4PCh3LS_Bbns-K_cj7Mpxf_uirsFWMurciYkJfkFYzKK16AfAJNA1q-DDVEnD8NiIO-uDzUNep_1pfwhMnc4n63YNFZY8RRs535KQ3lU4Y183-tzI-t937tEjTdyqS2wi39AK9n-u-J2956uSOmKe9h37cI--N9G4lxyEPQS19BKGjafFaYj9MPgZgYn_H_24KGKNcMknDCsAg3mM6OF-qcPY3xjCIDCTE7S4fWRNVDd0VlmrsArYw&h=_mri9de9euc_DjE-0qHvJT4QdcYJIEt2LrPkKcKCHqk
Cache-Control: no-cache
Date: Tue, 25 Nov 2025 21:40:14 GMT
Expires: -1
Location: https://management.azure.com/subscriptions/<sub_guid>/providers/Microsoft.DataProtection/locations/canadacentral/operationResults/NDA1ODVhNWEtYjk3Ny00ZmI5LTlhYWYtNjIwNTI3ODJiYjFkO2VjMDcwOWE2LWRmOTktNDUxZi1iMDA4LWQ4MDQxYjdlNWI4Nw==?api-version=2024-04-01&t=638997036148351494&c=MIIHpTCCBo2gAwIBAgITfwY2O9Fi5lSmLyOBKgAEBjY70TANBgkqhkiG9w0BAQsFADBEMRMwEQYKCZImiZPyLGQBGRYDR0JMMRMwEQYKCZImiZPyLGQBGRYDQU1FMRgwFgYDVQQDEw9BTUUgSW5mcmEgQ0EgMDIwHhcNMjUxMDIwMDQxMjEwWhcNMjYwNDE4MDQxMjEwWjBAMT4wPAYDVQQDEzVhc3luY29wZXJhdGlvbnNpZ25pbmdjZXJ0aWZpY2F0ZS5tYW5hZ2VtZW50LmF6dXJlLmNvbTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBAMHQm7ZXwmvz3bp_tBu0ZYNZMpMcRUJSHrld-E9b4RqS0ymmswmhM267lbAI2oCZNI82E4DZNNZApPYz9GqZQ2OYxE0G0E_dyvstulti8jSM29pm5HiR_hxNyzt_q0uSnmqQnWuxQMco30qz4XjZaQkJNBATBraJWnioUZtMKhrbXrMOMral9bW_6ba5EqNlc3iBHJ8x7dblRrVcao1gyCk1mEXz9vR8Ut0wZ-KLWuSUcbcS7CPzUuSYQ1eJdym4YYAqEQQM1z40lpqT6OlbqRLu3DC5dN2NbGHUPkHODMhUbKY3RnwHKdrKQbuDk16nRBjl2ZML6kXZGz_BMd-elV0CAwEAAaOCBJIwggSOMCcGCSsGAQQBgjcVCgQaMBgwCgYIKwYBBQUHAwEwCgYIKwYBBQUHAwIwPQYJKwYBBAGCNxUHBDAwLgYmKwYBBAGCNxUIhpDjDYTVtHiE8Ys-hZvdFs6dEoFghfmRS4WsmTQCAWQCAQcwggHaBggrBgEFBQcBAQSCAcwwggHIMGYGCCsGAQUFBzAChlpodHRwOi8vY3JsLm1pY3Jvc29mdC5jb20vcGtpaW5mcmEvQ2VydHMvQkwyUEtJSU5UQ0EwMS5BTUUuR0JMX0FNRSUyMEluZnJhJTIwQ0ElMjAwMig0KS5jcnQwVgYIKwYBBQUHMAKGSmh0dHA6Ly9jcmwxLmFtZS5nYmwvYWlhL0JMMlBLSUlOVENBMDEuQU1FLkdCTF9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3J0MFYGCCsGAQUFBzAChkpodHRwOi8vY3JsMi5hbWUuZ2JsL2FpYS9CTDJQS0lJTlRDQTAxLkFNRS5HQkxfQU1FJTIwSW5mcmElMjBDQSUyMDAyKDQpLmNydDBWBggrBgEFBQcwAoZKaHR0cDovL2NybDMuYW1lLmdibC9haWEvQkwyUEtJSU5UQ0EwMS5BTUUuR0JMX0FNRSUyMEluZnJhJTIwQ0ElMjAwMig0KS5jcnQwVgYIKwYBBQUHMAKGSmh0dHA6Ly9jcmw0LmFtZS5nYmwvYWlhL0JMMlBLSUlOVENBMDEuQU1FLkdCTF9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3J0MB0GA1UdDgQWBBQ7Uo6SICZvdsfQb4hxJfOTmnlMAzAOBgNVHQ8BAf8EBAMCBaAwggE1BgNVHR8EggEsMIIBKDCCASSgggEgoIIBHIZCaHR0cDovL2NybC5taWNyb3NvZnQuY29tL3BraWluZnJhL0NSTC9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3JshjRodHRwOi8vY3JsMS5hbWUuZ2JsL2NybC9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3JshjRodHRwOi8vY3JsMi5hbWUuZ2JsL2NybC9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3JshjRodHRwOi8vY3JsMy5hbWUuZ2JsL2NybC9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3JshjRodHRwOi8vY3JsNC5hbWUuZ2JsL2NybC9BTUUlMjBJbmZyYSUyMENBJTIwMDIoNCkuY3JsMIGdBgNVHSAEgZUwgZIwDAYKKwYBBAGCN3sBATBmBgorBgEEAYI3ewICMFgwVgYIKwYBBQUHAgIwSh5IADMAMwBlADAAMQA5ADIAMQAtADQAZAA2ADQALQA0AGYAOABjAC0AYQAwADUANQAtADUAYgBkAGEAZgBmAGQANQBlADMAMwBkMAwGCisGAQQBgjd7AwIwDAYKKwYBBAGCN3sEAjAfBgNVHSMEGDAWgBSuecJrXSWIEwb2BwnDl3x7l48dVTAdBgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIwDQYJKoZIhvcNAQELBQADggEBADp-OzkMAESDX27pXk3NL5_uU6RuCBkTjHnVxyWwp89arbWDWeL0FNU75N4hZMsK4FU9oUeatHrGk09j6u3o6qgzbyH7vWX9v5ijaUpNPeacRaMTpWvVZZLg6AERFF7oiplb-vCmLHZTMYbJH6dZI35unEEDgzD9n8QhuMBQ8dXhS145FHrAVc9nnNB25FQt29UfdpwrVFPQuEkgwKKCG8Qy5VDtGnceMRw2bnDwtYNngCgVmd2cPtwgizpflj_F8NURyxhnrQvuOwCkQavcJTq5RSXgEp3U471r3TZoEkKrAmcNF3Fyz5M0gOv6EQmdHg-lLMeFUmOVsXvWTlMjQQw&s=QKhrZbVUj5FTTOZsx5-IbNVBO9CyDjKoAxTQDI_K_6DS_oCeKNPamUsPwh6cEPaJdnfbDba4WtcGICrcYNDHWbLHrBMi9vquVUMsYofCR1Ws9uW2E2O9AdST8uiYaoguW-y2CsYnaWKytgtu-XGNjtt-sh8uiy7dfIlX4fyUA8QieYSEKJP5ql3d2ZdA2bhzCkfQzuTVR5Xa4_GpxYOCcmgnU3_RmvvtoEG6_ecwdE95qqsBH6ShLQekWLl0IUO1-91nmQ9LHLEK3_TaDuQk71-pC5fxnwBF5_Pfm5hbnaDK6uULMVhZagoF6rvUrLN0UXsA-nZjRnV9L8BjP_xiqA&h=UNrOuCMQjzZZrclJWVc0a1vEhEPK4PyYqbAJXlo7Idk
Pragma: no-cache
Retry-After: 30
Strict-Transport-Security: max-age=31536000; includeSubDomains
X-Cache: CONFIG_NOCACHE
X-Content-Type-Options: nosniff
X-Ms-Correlation-Request-Id: 909b11be-8d7c-7deb-e98e-e2ff08be4143
X-Ms-Operation-Identifier: appId=bcdae85a-6fc4-45f7-abf0-1cbdd6bd015a,tenantId=110fbd73-ac4a-4c62-8ca5-810aec218cd8,objectId=b788ac66-1dbc-4744-9f03-6f6585727daf/canadacentral/8067a637-c1d1-4e45-b370-b38a2e73a941
X-Ms-Ratelimit-Remaining-Subscription-Resource-Requests: 799
X-Ms-Request-Id: 71ea4e88-481a-462d-b65c-5f53a147223f
X-Ms-Routing-Request-Id: CANADACENTRAL:20251125T214014Z:71ea4e88-481a-462d-b65c-5f53a147223f
X-Msedge-Ref: Ref A: 708EA0D7E3644DD2861CEC0E74638F8F Ref B: CO6AA3150220039 Ref C: 2025-11-25T21:40:14Z
Content-Length: 0


2025/11/25 21:40:24 [DEBUG] AzureRM Request: 
GET /subscriptions/<sub_guid>/resourceGroups/acctest-dataprotection-251125213904888367/providers/Microsoft.DataProtection/backupVaults/acctest-dbv-251125213904888367?api-version=2024-04-01 HTTP/1.1
Host: management.azure.com
User-Agent: HashiCorp/go-azure-sdk (Go-http-Client/1.1 backupvaults/2024-04-01) HashiCorp Terraform/1.13.5 (+https://www.terraform.io) terraform-provider-azurerm/acc pid-222c6c49-1b0a-5959-a213-6608f9eb8820
Accept: application/json; charset=utf-8; IEEE754Compatible=false
Content-Type: application/json; charset=utf-8
Odata-Maxversion: 4.0
Odata-Version: 4.0
X-Ms-Correlation-Request-Id: 909b11be-8d7c-7deb-e98e-e2ff08be4143
Accept-Encoding: gzip


2025/11/25 21:40:24 [DEBUG] GET https://management.azure.com/subscriptions/<sub_guid>/resourceGroups/acctest-dataprotection-251125213904888367/providers/Microsoft.DataProtection/backupVaults/acctest-dbv-251125213904888367?api-version=2024-04-01
2025/11/25 21:40:25 [DEBUG] AzureRM Response for https://management.azure.com/subscriptions/<sub_guid>/resourceGroups/acctest-dataprotection-251125213904888367/providers/Microsoft.DataProtection/backupVaults/acctest-dbv-251125213904888367?api-version=2024-04-01: 
HTTP/2.0 404 Not Found
Content-Length: 218
Cache-Control: no-cache
Content-Language: en-US
Content-Type: application/json
Date: Tue, 25 Nov 2025 21:40:25 GMT
Expires: -1
Pragma: no-cache
Strict-Transport-Security: max-age=31536000; includeSubDomains
X-Cache: CONFIG_NOCACHE
X-Content-Type-Options: nosniff
X-Ms-Correlation-Request-Id: 909b11be-8d7c-7deb-e98e-e2ff08be4143
X-Ms-Error-Code: UserErrorGenericResourceNotFound
X-Ms-Ratelimit-Remaining-Subscription-Resource-Requests: 1099
X-Ms-Request-Id: fd688d94-344c-4540-999f-c8ad7119f519
X-Ms-Routing-Request-Id: CANADACENTRAL:20251125T214025Z:fd688d94-344c-4540-999f-c8ad7119f519
X-Msedge-Ref: Ref A: 74813081516B42EDA36D9AC17A730787 Ref B: CO6AA3150219019 Ref C: 2025-11-25T21:40:25Z

{"error":{"code":"UserErrorGenericResourceNotFound","message":"Azure resource does not exist. Please make sure required resources exist. If the issue persists, contact Microsoft support.","target":null,"details":null}}
2025-11-25T21:40:25.387Z [TRACE] sdk.helper_schema: Called downstream: tf_rpc=ApplyResourceChange tf_resource_type=azurerm_data_protection_backup_vault tf_req_id=db9771df-1470-97d0-3a46-2acf456fc331 tf_mux_provider="*schema.GRPCProviderServer" tf_provider_addr=registry.terraform.io/hashicorp/azurerm
2025-11-25T21:40:25.387Z [TRACE] sdk.proto: Received downstream response: diagnostic_error_count=0 tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_proto_version=5.10 tf_req_id=db9771df-1470-97d0-3a46-2acf456fc331 tf_rpc=ApplyResourceChange tf_resource_type=azurerm_data_protection_backup_vault diagnostic_warning_count=0 tf_req_duration_ms=11227
2025-11-25T21:40:25.387Z [TRACE] sdk.proto: Served request: tf_resource_type=azurerm_data_protection_backup_vault tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_proto_version=5.10 tf_req_id=db9771df-1470-97d0-3a46-2acf456fc331 tf_rpc=ApplyResourceChange
2025-11-25T21:40:25.390Z [TRACE] sdk.proto: Received request: tf_rpc=UpgradeResourceState tf_resource_type=azurerm_resource_group tf_req_id=69c2b00b-9791-7bef-374f-16c12e1f615a tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_proto_version=5.10
2025-11-25T21:40:25.390Z [TRACE] sdk.proto: Sending request downstream: tf_resource_type=azurerm_resource_group tf_req_id=69c2b00b-9791-7bef-374f-16c12e1f615a tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_proto_version=5.10 tf_rpc=UpgradeResourceState
2025-11-25T21:40:25.390Z [TRACE] sdk.mux: calling downstream server: tf_rpc=UpgradeResourceState tf_mux_provider="*schema.GRPCProviderServer"
2025-11-25T21:40:25.390Z [TRACE] sdk.helper_schema: Upgrading JSON state: tf_req_id=69c2b00b-9791-7bef-374f-16c12e1f615a tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_mux_provider="*schema.GRPCProviderServer" tf_rpc=UpgradeResourceState tf_resource_type=azurerm_resource_group
2025-11-25T21:40:25.390Z [TRACE] sdk.proto: Received downstream response: tf_req_duration_ms=0 diagnostic_error_count=0 tf_rpc=UpgradeResourceState tf_resource_type=azurerm_resource_group tf_req_id=69c2b00b-9791-7bef-374f-16c12e1f615a tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_proto_version=5.10 diagnostic_warning_count=0
2025-11-25T21:40:25.390Z [TRACE] sdk.proto: Served request: tf_req_id=69c2b00b-9791-7bef-374f-16c12e1f615a tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_proto_version=5.10 tf_rpc=UpgradeResourceState tf_resource_type=azurerm_resource_group
2025-11-25T21:40:25.390Z [TRACE] sdk.proto: Received request: tf_rpc=ApplyResourceChange tf_resource_type=azurerm_resource_group tf_proto_version=5.10 tf_req_id=a6ac521f-783f-8884-e726-edc8d0fde017 tf_provider_addr=registry.terraform.io/hashicorp/azurerm
2025-11-25T21:40:25.390Z [TRACE] sdk.proto: Sending request downstream: tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_rpc=ApplyResourceChange tf_resource_type=azurerm_resource_group tf_proto_version=5.10 tf_req_id=a6ac521f-783f-8884-e726-edc8d0fde017
2025-11-25T21:40:25.390Z [TRACE] sdk.mux: calling downstream server: tf_rpc=ApplyResourceChange tf_mux_provider="*schema.GRPCProviderServer"
2025-11-25T21:40:25.390Z [TRACE] sdk.helper_schema: Calling downstream: tf_mux_provider="*schema.GRPCProviderServer" tf_req_id=a6ac521f-783f-8884-e726-edc8d0fde017 tf_resource_type=azurerm_resource_group tf_rpc=ApplyResourceChange tf_provider_addr=registry.terraform.io/hashicorp/azurerm
2025/11/25 21:40:35 [DEBUG] AzureRM Request: 
GET /subscriptions/<sub_guid>/resourceGroups/acctest-dataprotection-251125213904888367/resources?%24expand=provisioningState&%24top=10&api-version=2023-07-01 HTTP/1.1
Host: management.azure.com
User-Agent: HashiCorp/go-azure-sdk (Go-http-Client/1.1 resourcegroups/2023-07-01) HashiCorp Terraform/1.13.5 (+https://www.terraform.io) terraform-provider-azurerm/acc pid-222c6c49-1b0a-5959-a213-6608f9eb8820
Accept: application/json; charset=utf-8; IEEE754Compatible=false
Content-Type: application/json; charset=utf-8
Odata-Maxversion: 4.0
Odata-Version: 4.0
Accept-Encoding: gzip


2025/11/25 21:40:35 [DEBUG] GET https://management.azure.com/subscriptions/<sub_guid>/resourceGroups/acctest-dataprotection-251125213904888367/resources?%24expand=provisioningState&%24top=10&api-version=2023-07-01
2025/11/25 21:40:35 [DEBUG] AzureRM Response for https://management.azure.com/subscriptions/<sub_guid>/resourceGroups/acctest-dataprotection-251125213904888367/resources?%24expand=provisioningState&%24top=10&api-version=2023-07-01: 
HTTP/2.0 200 OK
Content-Length: 391
Cache-Control: no-cache
Content-Type: application/json; charset=utf-8
Date: Tue, 25 Nov 2025 21:40:35 GMT
Expires: -1
Pragma: no-cache
Strict-Transport-Security: max-age=31536000; includeSubDomains
X-Cache: CONFIG_NOCACHE
X-Content-Type-Options: nosniff
X-Ms-Correlation-Request-Id: 94e07c01-5242-4c8e-a06c-a85aa482f64c
X-Ms-Ratelimit-Remaining-Subscription-Global-Reads: 3749
X-Ms-Ratelimit-Remaining-Subscription-Reads: 249
X-Ms-Request-Id: 94e07c01-5242-4c8e-a06c-a85aa482f64c
X-Ms-Routing-Request-Id: CANADACENTRAL:20251125T214035Z:94e07c01-5242-4c8e-a06c-a85aa482f64c
X-Msedge-Ref: Ref A: 12DFA3540C3C486880C80828FD3F190A Ref B: CO6AA3150217027 Ref C: 2025-11-25T21:40:35Z

{"value":[{"id":"/subscriptions/<sub_guid>/resourceGroups/acctest-dataprotection-251125213904888367/providers/Microsoft.DataProtection/BackupVaults/acctest-dbv-251125213904888367","name":"acctest-dbv-251125213904888367","type":"Microsoft.DataProtection/BackupVaults","location":"canadacentral","identity":{"type":"None"},"provisioningState":"Succeeded","tags":{}}]}
```

Note that Azure responds with a `202 Accepted` as before, but now the provider waits 10 seconds and polls to see if the resource is gone (hooray!), Azure replies with a `404` (so it's gone right?) but then the subsequent call to the `/resources?$expand=provisioningState&$top=10&api-version=2023-07-01` API replies with the vault resource continuing to exist.

A subsequent `terraform destroy` succeeds in destroying the RG and indeed if you look in the portal there's nothing in that RG after a bit.

So the question is, what's changed?

And/or, can we do something in the code, when we know there's something weird like this occurring with the backend API, that forces a certain number of polling cycles (say, 3) no matter what before continuing?

### Steps to Reproduce

1. `terraform apply`
2. `terraform destroy` 

### Important Factoids

_No response_

### References

See above.

--- a/go.mod
+++ b/go.mod
@@ -17,8 +17,8 @@ require (
 	github.com/google/go-cmp v0.7.0
 	github.com/google/uuid v1.6.0
 	github.com/hashicorp/go-azure-helpers v0.75.1
-	github.com/hashicorp/go-azure-sdk/resource-manager v0.20251126.1220923
-	github.com/hashicorp/go-azure-sdk/sdk v0.20251126.1220923
+	github.com/hashicorp/go-azure-sdk/resource-manager v0.20251202.1181053
+	github.com/hashicorp/go-azure-sdk/sdk v0.20251202.1181053
 	github.com/hashicorp/go-cty v1.5.0
 	github.com/hashicorp/go-hclog v1.6.3
 	github.com/hashicorp/go-multierror v1.1.1

--- a/internal/services/resource/resource_group_resource.go
+++ b/internal/services/resource/resource_group_resource.go
@@ -4,6 +4,7 @@
 package resource
 
 import (
+	"context"
 	"fmt"
 	"log"
 	"time"
@@ -181,9 +182,12 @@ func resourceResourceGroupDelete(d *pluginsdk.ResourceData, meta interface{}) er
 	// conditionally check for nested resources and error if they exist
 	if meta.(*clients.Client).Features.ResourceGroup.PreventDeletionIfContainsResources {
 		// Resource groups sometimes hold on to resource information after the resources have been deleted. We'll retry this check to account for that eventual consistency.
+		deletePollerContext, deletePollerCancel := context.WithTimeout(context.Background(), 10*time.Minute)
+		defer deletePollerCancel()
+
 		pollerType := custompollers.NewResourceGroupPreventDeletePoller(client, *id)
-		poller := pollers.NewPoller(pollerType, 10*time.Second, pollers.DefaultNumberOfDroppedConnectionsToAllow)
-		if err := poller.PollUntilDone(ctx); err != nil {
+		poller := pollers.NewRetryOnErrorPoller(pollerType, 10*time.Second, pollers.DefaultNumberOfDroppedConnectionsToAllow, true)
+		if err := poller.PollUntilDone(deletePollerContext); err != nil {
 			return err
 		}
 	}

